ui/src/router/components/form/components/ensembler_config/typeOptions.js

```diff
@@ -62,11 +75,10 @@ export const ensemblerTypeOptions = (engineProps) => {
     return typeOptions.filter((o) => o.value !== "standard");
   }
   // Ensembler must be selected when there is an experiment engine
-  const ensemblerOptions = typeOptions.filter((o) => o.value !== "nop");
-  if (
-    !engineProps?.standard_experiment_manager_config
-      ?.experiment_selection_enabled
-  ) {
+  const ensemblerOptions = typeOptions.filter(
+    (o) => o.value !== "nop" && (o.expenginetype === engineProps.type || o.expenginetype === undefined)
```

Comment:
This additional condition selects the standard ensembler `typeOption` with the correct `expenginetype`. If a `typeOption` does not have the `expenginetype` field set (which implicitly means it's not a standard ensembler), then it will not be removed.

Note: Just noticed the 2 subconditions here `(o.expenginetype === engineProps.type || o.expenginetype === undefined)` can be inverted for slightly better computation 😅